Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1482,6 +1505,26 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
                 GstClockTime time;
                 gst_structure_get(structure, "uri", G_TYPE_STRING, &uri.outPtr(), "fragment-download-time", GST_TYPE_CLOCK_TIME, &time, nullptr);
                 GST_TRACE("Fragment %s download time %" GST_TIME_FORMAT, uri.get(), GST_TIME_ARGS(time));
+            } else if (isMediaDiskCacheDisabled() && gst_structure_has_name(structure, "webkit-network-statistics")) {
+                guint64 networkReadPosition = 0;
+                guint64 httpResponseTotalSize = 0;
+
+                if (gst_structure_get(structure, "read-position", G_TYPE_UINT64, &networkReadPosition, "size", G_TYPE_UINT64, &httpResponseTotalSize, nullptr)) {
+                    GST_DEBUG_OBJECT(pipeline(), "Updated network read position %" G_GUINT64_FORMAT ", size: %" G_GUINT64_FORMAT, networkReadPosition, httpResponseTotalSize);
+
+                    MediaTime mediaDuration = durationMediaTime();
+                    // Update maxTimeLoaded only if the media duration is
+                    // available. Otherwise we can't compute it.
+                    if (mediaDuration && httpResponseTotalSize) {
+                        double fillStatus = 100.0 * (static_cast<double>(networkReadPosition) / static_cast<double>(httpResponseTotalSize));
+
+                        if (fillStatus == 100.0)
+                            m_maxTimeLoaded = mediaDuration;
+                        else
+                            m_maxTimeLoaded = MediaTime(fillStatus * static_cast<double>(toGstUnsigned64Time(mediaDuration)) / 100, GST_SECOND);
```

Comment:
double fillStatus = (static_cast<double>(networkReadPosition) / static_cast<double>(httpResponseTotalSize));
m_maxTimeLoaded = MediaTime(fillStatus * static_cast<double>(toGstUnsigned64Time(mediaDuration)), GST_SECOND);

wouldn't this be simpler? no need to multiply by 100 just to divide a few lines below, also special case for 100% is not needed. you're not logging the percentages anyway